File: Sequence Valley/String Matching/Q2. Rotate String/Q2_RotateString.java
class Solution {
    public boolean rotateString(String s, String goal) {
        // First check: lengths must be equal
        if (s.length() != goal.length()) {
            return false;
        }

        // Concatenate s+s → contains all rotations
        String doubled = s + s;

        // Check if goal is substring of doubled string
        return doubled.contains(goal);
    }
}

File: Sequence Valley/String Matching/Q2. Rotate String/Q2_RotateString.py
class Solution(object):                 # Define a class named Solution
    def rotateString(self, s, goal):    # Define function to check rotation
        """
        :type s: str                    # s is a string
        :type goal: str                 # goal is a string
        :rtype: bool                    # return value is True or False
        """
        if len(s) != len(goal):         # If lengths are different, rotation is impossible
            return False                # Return False

        return goal in s + s            # Check if goal exists in s+s (all rotations)
